packages/block-library/src/cover/edit/index.js

```diff
@@ -285,18 +341,42 @@ function CoverEdit( {
 	};
 
 	const toggleUseFeaturedImage = async () => {
-		const isDarkSetting = await ( useFeaturedImage
-			? getCoverIsDark( undefined, dimRatio, overlayColor.color )
-			: getCoverIsDark( mediaUrl, dimRatio, overlayColor.color ) );
+		const newUseFeaturedImage = ! useFeaturedImage;
+
+		const averageBackgroundColor = newUseFeaturedImage
+			? await getMediaColor( mediaUrl )
+			: DEFAULT_AVERAGE_COLOR;
+
+		const newOverlayColor = ! isUserOverlayColor
+			? averageBackgroundColor
+			: overlayColor.color;
+
+		if ( ! isUserOverlayColor ) {
+			if ( newUseFeaturedImage ) {
+				setOverlayColor( newOverlayColor );
+			} else {
+				setOverlayColor( undefined );
+			}
+
+			// Make undo revert the next setAttributes and the previous setOverlayColor.
+			__unstableMarkNextChangeAsNotPersistent();
+		}
```

Comment:
Similar to clearing media that I mentioned [above](https://github.com/WordPress/gutenberg/pull/54054#discussion_r1330883005), when I was playing around with test cases, it felt more natural for the overlay color to be reset when toggling off the featured image if the user hadn't specified their own overlay color.

1. Create cover block with featured image. (Do not manually set an overlay color.)
2. Toggle off the featured image.
3. What should happen?

If it should revert to using the default CSS black overlay, nothing needs to be changed.

However, if you think it should retain the computed color even with `isUserOverlayColor` unset, then you can apply this suggestion.

```suggestion
		if ( ! isUserOverlayColor && newUseFeaturedImage) {
			setOverlayColor( newOverlayColor );

			// Make undo revert the next setAttributes and the previous setOverlayColor.
			__unstableMarkNextChangeAsNotPersistent();
		}
```

Reply:
Same as the case of setting media, using a featured image with a system set color, means removing featured image should remove the color.